lib/logger/utils.ts

```diff
@@ -32,3 +35,67 @@ export class ErrorStream extends Stream {
     return this._errors;
   }
 }
+
+function sanitizeValue(value: any, seen = new WeakMap()) {
+  if (Array.isArray(value)) {
+    const length = value.length;
+    const arrayResult = Array(length);
+    seen.set(value, arrayResult);
+    for (let idx = 0; idx < length; idx += 1) {
+      const val = value[idx];
+      arrayResult[idx] = seen.has(val)
+        ? seen.get(val)
+        : sanitizeValue(val, seen);
+    }
+    return arrayResult;
+  }
+
+  const valueType = typeof value;
+
+  if (value != null && valueType !== 'function' && valueType === 'object') {
+    const objectResult: Record<string, any> = {};
+    seen.set(value, objectResult);
+    for (const [key, val] of Object.entries(value)) {
+      objectResult[key] = seen.has(val as object)
```

Comment:
@zharinov Sorry, but here is another one. Can you please check all other too (eg L60)